tests/Unit/PointwiseFunctions/AnalyticSolutions/EinsteinSolutions/VerifyEinsteinSolution.cpp

```diff
@@ -0,0 +1,284 @@
+      d4_psi.get(0, b, c) = -get(lapse) * pi.get(b, c);
+      for (size_t k = 0; k < 3; ++k) {
+        d4_psi.get(0, b, c) += shift.get(k) * phi.get(k, b, c);
+        d4_psi.get(k + 1, b, c) = phi.get(k, b, c);
+      }
+    }
+  }
+
+  // Compute derived spacetime quantities
+  const auto upper_psi =
+      gr::inverse_spacetime_metric(lapse, shift, upper_spatial_metric);
+  const auto christoffel_first_kind = compute_christoffel_first_kind(d4_psi);
+  const auto christoffel_second_kind =
+      raise_or_lower_first_index(christoffel_first_kind, upper_psi);
+  const auto trace_christoffel_first_kind =
+      trace_last_indices(christoffel_first_kind, upper_psi);
+  const auto normal_one_form =
+      gr::spacetime_normal_one_form<3, Frame::Inertial>(lapse);
+  const auto normal_vector = gr::spacetime_normal_vector(lapse, shift);
+
+  // Test ADM evolution equation gives zero
+  auto dt_g_adm = make_with_value<tnsr::ii<DataVector, 3>>(x, 0.0);
+  {
+    auto ex_curv = gr::extrinsic_curvature(lapse, shift, d_shift, g, dt_g, d_g);
+    for (size_t i = 0; i < 3; ++i) {
+      for (size_t j = i; j < 3; ++j) {  // Symmetry
+        dt_g_adm.get(i, j) -= 2.0 * get(lapse) * ex_curv.get(i, j);
+        for (size_t k = 0; k < 3; ++k) {
+          dt_g_adm.get(i, j) += shift.get(k) * d_g.get(k, i, j) +
+                                g.get(k, i) * d_shift.get(j, k) +
+                                g.get(k, j) * d_shift.get(i, k);
+        }
+      }
+    }
+    for (size_t i = 0; i < 3; ++i) {
+      for (size_t j = 0; j < 3; ++j) {
+        CHECK_ITERABLE_APPROX(dt_g_adm.get(i, j), zero);
+      }
+    }
+  }
+
+  // Test 1-index constraint
+  auto C_1 = make_with_value<tnsr::a<DataVector, 3>>(x, 0.0);
+  for (size_t a = 0; a < 4; ++a) {
+    C_1.get(a) = gauge_function.get(a);
+    for (size_t i = 0; i < 3; ++i) {
+      for (size_t j = 0; j < 3; ++j) {
+        C_1.get(a) += upper_spatial_metric.get(i, j) * phi.get(i, j + 1, a);
+      }
+    }
+    for (size_t b = 0; b < 4; ++b) {
+      C_1.get(a) += normal_vector.get(b) * pi.get(b, a);
+      for (size_t c = 0; c < 4; ++c) {
+        C_1.get(a) -=
+            0.5 * normal_one_form.get(a) * pi.get(b, c) * upper_psi.get(b, c);
+        if (a > 0) {
+          C_1.get(a) -= 0.5 * phi.get(a - 1, b, c) * upper_psi.get(b, c);
+        }
+        for (size_t i = 0; i < 3; ++i) {
+          C_1.get(a) -= 0.5 * normal_one_form.get(a) *
+                        normal_vector.get(i + 1) * phi.get(i, b, c) *
+                        upper_psi.get(b, c);
+        }
+      }
+    }
+  }
+  for (size_t a = 0; a < 4; ++a) {
+    for (size_t s = 0; s < data_size; ++s) {
+      CHECK(C_1.get(a)[s] == numerical_approx(0.0));
+    }
+  }
+
+  // Constraint-damping parameters: Set to arbitrary values.
+  // gamma1 = 0 and -1 are special, so don't choose that one.
+  const auto gamma0 = make_with_value<Scalar<DataVector>>(x, 1.0);
+  const auto gamma1 = make_with_value<Scalar<DataVector>>(x, 1.0);
```

Comment:
`const auto& gamma1 = gamma0;`

Reply:
Why use a reference? I think this is correct as is since the RHS will be temporary.

Reply:
:+1:

Reply:
It would be an optimization, as it avoids an allocation, but since this is just a test that shouldn't matter.  It's probably clearer as is, so I'd leave it.